pkg/common-controller/snapshot_controller.go

```diff
@@ -1489,14 +1544,21 @@ func (ctrl *csiSnapshotCommonController) setAnnVolumeSnapshotBeingDeleted(conten
 	// Set AnnVolumeSnapshotBeingDeleted if it is not set yet
 	if !metav1.HasAnnotation(content.ObjectMeta, utils.AnnVolumeSnapshotBeingDeleted) {
 		klog.V(5).Infof("setAnnVolumeSnapshotBeingDeleted: set annotation [%s] on content [%s].", utils.AnnVolumeSnapshotBeingDeleted, content.Name)
+		var patches []utils.PatchOp
 		metav1.SetMetaDataAnnotation(&content.ObjectMeta, utils.AnnVolumeSnapshotBeingDeleted, "yes")
+		patches = append(patches, utils.PatchOp{
+			Op:    "replace",
+			Path:  "/metadata/annotations",
```

Comment:
Does this work if there were no annotations initially?

Reply:
Yes, tested this here:
```
grant@colo-70-0-0-129:~/workspace/kubeup$ k get volumesnapshotcontent snapcontent-52f8092e-148f-4747-8f57-f1f21eaed3d4 -o yaml | grep annotations
grant@colo-70-0-0-129:~/workspace/kubeup$ k patch volumesnapshotcontent snapcontent-52f8092e-148f-4747-8f57-f1f21eaed3d4 --type='json' -p='[{"op": "replace", "path": "/metadata/annotations", "value": {"snapshot.storage.kubernetes.io/volumesnapshot-being-deleted": "yes"}}]'
volumesnapshotcontent.snapshot.storage.k8s.io/snapcontent-52f8092e-148f-4747-8f57-f1f21eaed3d4 patched
grant@colo-70-0-0-129:~/workspace/kubeup$ k get volumesnapshotcontent snapcontent-52f8092e-148f-4747-8f57-f1f21eaed3d4 -o yaml | grep -A 2 -B 2 annotations
kind: VolumeSnapshotContent
metadata:
  annotations:
    snapshot.storage.kubernetes.io/volumesnapshot-being-deleted: "yes"
  creationTimestamp: "2021-09-24T02:15:45Z"
--
    fieldsV1:
      f:metadata:
        f:annotations:
          .: {}
          f:snapshot.storage.kubernetes.io/volumesnapshot-being-deleted: {}

```